scripts/simple-scylla-cluster-manager.py

```diff
@@ -0,0 +1,776 @@
+    sin = os.path.join(cluster_directory, FIFO_IN)
+    sout = os.path.join(cluster_directory, FIFO_OUT)
+    return sin, sout
+
+
+class SupervisorClient:
+    """Local client, for the supervisor
+
+    Allows initializing a connection and sending commands to the supervisor.
+    """
+
+    def __init__(self, cluster_directory):
+        """Initialize the client for a supervisor running in the specified cluster directory"""
+        self.supervisor_in_fifo, self.supervisor_out_fifo = _get_fifo_paths(cluster_directory)
+
+    def send_command(self, argv):
+        """Send commands to the supervisor
+
+        Only node commands are supported, e.g. commands ending with the *-node
+        prefix as opposed to those with a *-cluster prefix.
+        Pass the full command-line in the argv parameter, e.g.:
+
+            client = sscm.SupervisorClient("/path/to/cluster_dir")
+            client.send_command(["start-node", "-i1"])
+
+        """
+        if not os.path.exists(self.supervisor_in_fifo) or not os.path.exists(self.supervisor_out_fifo):
+            raise RuntimeError(f"Failed to find supervisor input/output FIFO at {self.supervisor_in_fifo} and {self.supervisor_out_fifo}, is the supervisor running?")
+
+        with open(self.supervisor_in_fifo, "w") as pout:
+            with open(self.supervisor_out_fifo, "r") as pin:
+                pout.write(' '.join(argv) + '\n')
+                pout.flush()
+
+                res = json.loads(pin.readline())
+                if res["success"]:
+                    if res["content"]:
+                        return res["content"]
+                    return None
+
+                raise RuntimeError(f"Operation failed, supervisor responed with: {res['content']}")
+
+
+def _start_supervisor(argv0, cluster_directory):
+    if os.path.exists(os.path.join(cluster_directory, PID_FILE)):
+        raise RuntimeError(f"{PID_FILE} already exists, is the supervisor running?")
+
+    sin, sout = _get_fifo_paths(cluster_directory)
+
+    os.mkfifo(sin)
+    os.mkfifo(sout)
+
+    with open(os.path.join(cluster_directory, LOG_FILE), "w") as supervisor_log:
+        supervisor_process = subprocess.Popen(
+                [sys.executable, os.path.abspath(argv0), "supervisor"],
+                stdin = subprocess.DEVNULL,
+                stdout = supervisor_log,
+                stderr = subprocess.STDOUT,
+                cwd = os.path.abspath(cluster_directory))
+
+    supervisor_client = SupervisorClient(cluster_directory)
+
+    # The supervisor can only successfully open the pipes if they are also
+    # opened here with the opposite mode. Send a "check" command to coordinate
+    # opening the pipes. This command also serves the purpose of checking if the
+    # supervisor was successfully initialized.
+    supervisor_client.send_command(["check"])
+
+    return supervisor_client
+
+
+def _maybe_start_nodes(nodes, start_all, supervisor_client):
+    if not start_all or nodes == 0:
```

Comment:
it's fine if `nodes` <= 0. in that case `range(1, nodes+1)` would generate nothing.

Reply:
Yes, but I like the clarity of an early return. I also don't have to watch for what happens after the loop and so on.